hermes/bin/src/runtime_extensions/hermes/http_gateway/routing.rs

```diff
@@ -115,65 +109,151 @@ pub(crate) async fn router(
     Ok(response)
 }
 
-/// Routes HTTP requests to WASM modules or static file handlers
-///
-/// Converts incoming HTTP requests into structured events for WASM processing,
-/// preserving full URLs including query parameters for accurate forwarding.
-///
-/// ## Routing
-/// - `/api/*` → WASM modules via event queue
-/// - Valid paths → Static file system
-/// - Invalid paths → HTTP 404
+/// HTTP error response generator
+pub(crate) fn error_response<B>(err: impl Into<String>) -> anyhow::Result<Response<B>>
+where B: Body + From<String> {
+    Response::builder()
+        .status(StatusCode::INTERNAL_SERVER_ERROR)
+        .body(err.into().into())
+        .map_err(Into::into)
+}
+
+/// HTTP not found response generator
+fn not_found<B>() -> anyhow::Result<Response<B>>
+where B: Body + From<Vec<u8>> {
+    let response = Response::builder()
+        .status(StatusCode::NOT_FOUND)
+        .body(b"Not Found".to_vec().into())?;
+    add_security_headers(response)
+}
+
+/// Extractor that resolves the hostname of the request.
+/// Hostname is resolved through the Host header
+pub(crate) fn host_resolver(headers: &HeaderMap) -> anyhow::Result<(ApplicationName, Hostname)> {
+    let host = headers
+        .get("Host")
+        .and_then(|h| h.to_str().ok())
```

Comment:
We could use predeclared [constants](https://docs.rs/restless/latest/restless/clients/hyper/http/header/index.html#:~:text=vacant%20or%20occupied.-,Constants,-%C2%A7) for headers.
```suggestion
        .get(HOST)
        .and_then(|h| h.to_str().ok())
```